plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeParquetStatisticsUtils.java

```diff
@@ -71,6 +92,119 @@ public static boolean hasInvalidStatistics(Collection<ColumnChunkMetaData> metad
                         (!metadata.getStatistics().hasNonNullValue() && metadata.getStatistics().getNumNulls() != metadata.getValueCount()));
     }
 
+    @Nullable
+    public static Object jsonValueToTrinoValue(Type type, @Nullable Object jsonValue)
+    {
+        if (jsonValue == null) {
+            return null;
+        }
+
+        if (type == SMALLINT || type == TINYINT || type == INTEGER) {
+            return (long) (int) jsonValue;
+        }
+        if (type == BIGINT) {
+            return (long) (int) jsonValue;
+        }
+        if (type == REAL) {
+            return (long) floatToRawIntBits((float) (double) jsonValue);
+        }
+        if (type == DOUBLE) {
+            return jsonValue;
+        }
+        if (type instanceof DecimalType) {
+            BigDecimal decimal;
+            checkArgument(jsonValue instanceof String || jsonValue instanceof Double, "Value must be instance of String or Double");
+            if (jsonValue instanceof String) {
+                decimal = new BigDecimal((String) jsonValue);
+            }
+            else {
+                decimal = BigDecimal.valueOf((double) jsonValue);
+            }
+
+            if (isShortDecimal(type)) {
+                return Decimals.encodeShortScaledValue(decimal, ((DecimalType) type).getScale());
+            }
+            return Decimals.encodeScaledValue(decimal, ((DecimalType) type).getScale());
+        }
+        if (type instanceof VarcharType) {
+            return jsonValue;
+        }
+        if (type == DateType.DATE) {
+            return LocalDate.parse((String) jsonValue).toEpochDay();
+        }
+        if (type == TIMESTAMP_MILLIS) {
+            return Instant.parse((String) jsonValue).toEpochMilli() * MICROSECONDS_PER_MILLISECOND;
+        }
+        if (type instanceof RowType rowType) {
+            Map<?, ?> values = (Map<?, ?>) jsonValue;
+            List<Type> fieldTypes = rowType.getTypeParameters();
+            BlockBuilder blockBuilder = new RowBlockBuilder(fieldTypes, null, 1);
+            BlockBuilder singleRowBlockWriter = blockBuilder.beginBlockEntry();
+            for (int i = 0; i < values.size(); ++i) {
+                Type fieldType = fieldTypes.get(i);
+                Object fieldValue = jsonValueToTrinoValue(fieldType, values.get(rowType.getFields().get(i).getName().orElseThrow()));
```

Comment:
Should we validate that `values` contains no other entries than the ones we expected?